Tests/StencilTests/VariableSpec.swift

```diff
@@ -188,6 +188,107 @@ func testVariable() {
       let result = try variable.resolve(context) as? Int
       try expect(result) == 2
     }
+
+    $0.describe("Indirection") {
+      $0.it("can resolve an indirect variable via reflection") {
+        try context.push(dictionary: ["property": "name"]) {
+          let variable = Variable("article.author[property]")
+          let result = try variable.resolve(context) as? String
+          try expect(result) == "Kyle"
+        }
+      }
+
+      $0.it("can resolve an item from an array via it's indirect index") {
+        try context.push(dictionary: ["property": 0]) {
+          let variable = Variable("contacts[property]")
+          let result = try variable.resolve(context) as? String
+          try expect(result) == "Katie"
+        }
+      }
+
+      $0.it("can resolve an item from an array via unknown index") {
+        try context.push(dictionary: ["property": 5]) {
+          let variable = Variable("contacts[property]")
+          let result = try variable.resolve(context) as? String
+          try expect(result).to.beNil()
```

Comment:
I think django/jinja just return `nil`s, like here. It's up to the developer to check for empty cases and handle these accordingly. How would you otherwise check if something exists or not, without triggering an exception?

Reply:
I see it like that. Lookup function should return nil if variable is not defined in the context, but if it is array subscript or type introspection it should throw. But even in the first case I don't think nil lookup result should be rendered as empty strings, it should be exception. In Jinja I believe one can check if variable is not nil using `is defined` (`{% if loop.previtem is defined and value > loop.previtem %}`). But I'm not very well familiar with Jinja and Django to advocate for that and it is not related to this PR so lets leave it for another time.

Reply:
Quote from Jinja template docs:
> If a variable or attribute does not exist, you will get back an undefined value. What you can do with that kind of value depends on the application configuration: the default behavior is to evaluate to an empty string if printed or iterated over, and to fail for every other operation.

So it returns an invalid value (doesn't throw), even for subscripts, that can be printed, iterated (and tested in `if`s I assume), but other operations will fail (exception) for it.